src/components/Api.js

```diff
@@ -0,0 +1,136 @@
+export default class Api {
+  constructor(options) {
+    //constructor body
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.all(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    })
+      .then(this._checkResponse)
+      .catch((err) => {
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`   after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.